tests/unit_tests/net.cpp

```diff
@@ -936,6 +936,35 @@ TEST(get_network_address, ipv4subnet)
     EXPECT_STREQ("12.34.0.0/16", address->str().c_str());
 }
 
+TEST(get_network_address, host_and_port)
+{
+#   define NA_HOST_AND_PORT_TEST(addr, exp_host, exp_port)            \
+        do {                                                          \
+            std::string host{"xxxxx"};                                \
+            std::string port{"xxxxx"};                                \
+            net::get_network_address_host_and_port(addr, host, port); \
+            EXPECT_EQ(exp_host, host);                                \
+            EXPECT_EQ(exp_port, port);                                \
+        } while (0);                                                  \
+
+    // IPv4
+    NA_HOST_AND_PORT_TEST("9.9.9.9", "9.9.9.9", "xxxxx");
+    NA_HOST_AND_PORT_TEST("9.9.9.9:18081", "9.9.9.9", "18081");
+
+    // IPv6
+    NA_HOST_AND_PORT_TEST("::ffff", "::ffff", "xxxxx");
+    NA_HOST_AND_PORT_TEST("[::ffff]", "::ffff", "xxxxx");
+    NA_HOST_AND_PORT_TEST("[::ffff]:00231", "::ffff", "00231");
+    NA_HOST_AND_PORT_TEST("ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff", "ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff", "xxxxx");
+    NA_HOST_AND_PORT_TEST("[7777:7777:7777:7777:7777:7777:7777:7777]", "7777:7777:7777:7777:7777:7777:7777:7777", "xxxxx");
+    NA_HOST_AND_PORT_TEST("[7777:7777:7777:7777:7777:7777:7777:7777]:48080", "7777:7777:7777:7777:7777:7777:7777:7777", "48080");
+
+    // DNS names
+    NA_HOST_AND_PORT_TEST("localhost", "localhost", "xxxxx");
```

Comment:
These could be in a separate `TEST(get_network_address, hostname)` test function. Or just merge everything into one function, its not that much work. Again this helps with quick identification and filtering.

Although, I'm not sure that these tests matter that much given that there is nothing special for `onion` or `i2p` addresses. This isn't strictly an internal blackbox either; `get_network_address_and_port` has `std::string` out parameters that don't provide anything unique for those special TLDs.